process/smartContract/sovereignSCProcess.go

```diff
@@ -0,0 +1,649 @@
+package smartContract
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"sort"
+	"strings"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/smartContractResult"
+	vmData "github.com/multiversx/mx-chain-core-go/data/vm"
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/errors"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/sharding"
+	"github.com/multiversx/mx-chain-go/state"
+	"github.com/multiversx/mx-chain-go/storage"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type sovereignSCProcessor struct {
+	blockChainHook     process.BlockChainHookHandler
+	shardCoordinator   sharding.Coordinator
+	pubkeyConv         core.PubkeyConverter
+	hasher             hashing.Hasher
+	marshalizer        marshal.Marshalizer
+	txTypeHandler      process.TxTypeHandler
+	accounts           state.AccountsAdapter
+	argsParser         process.ArgumentsParser
+	txLogsProcessor    process.TransactionLogProcessor
+	vmOutputCacher     storage.Cacher
+	esdtTransferParser vmcommon.ESDTTransferParser
+	wasmVMChangeLocker common.Locker
+	vmContainer        process.VirtualMachinesContainer
+	scrForwarder       process.IntermediateTransactionHandler
+}
+
+// ProcessSmartContractResult updates the account state from the smart contract result
+func (sc *sovereignSCProcessor) ProcessSmartContractResult(scr *smartContractResult.SmartContractResult) (vmcommon.ReturnCode, error) {
+	if check.IfNil(scr) {
+		return 0, process.ErrNilSmartContractResult
+	}
+
+	log.Trace("sovereignSCProcessor.ProcessSmartContractResult()", "sender", scr.GetSndAddr(), "receiver", scr.GetRcvAddr(), "data", string(scr.GetData()))
+
+	var err error
+	returnCode := vmcommon.UserError
+	txHash, err := core.CalculateHash(sc.marshalizer, sc.hasher, scr)
+	if err != nil {
+		log.Debug("CalculateHash error", "error", err)
+		return returnCode, err
+	}
+
+	dstAcc, err := sc.getAccountFromAddress(scr.RcvAddr)
+	if err != nil {
+		return returnCode, err
+	}
+	sndAcc, err := sc.getAccountFromAddress(scr.SndAddr)
+	if err != nil {
+		return returnCode, err
+	}
+
+	if check.IfNil(dstAcc) {
+		err = process.ErrNilSCDestAccount
+		return returnCode, err
+	}
+
+	if !bytes.Equal(sndAcc.AddressBytes(), core.ESDTSCAddress) {
+		return returnCode, fmt.Errorf("%w, expected ESDTSCAddress", errInvalidSenderAddress)
+	}
+
+	snapshot := sc.accounts.JournalLen()
+	process.DisplayProcessTxDetails(
+		"ProcessSmartContractResult: receiver account details",
+		dstAcc,
+		scr,
+		txHash,
+		sc.pubkeyConv,
+	)
```

Comment:
lines 46 - 85 should be moved to a common code. also we do not care about sndAcc -> as that is the ESDTSCAddress

Reply:
Right,
Did the changes as requested